### PR TITLE
NEO address

### PR DESCRIPTION
**Pull request by ArtemGoryunov**
## Description

Added NEO address with unit tests

## Testing instructions

Run unit tests

## Checklist

- [x] Prefix PR title with `[WIP]` if necessary.
- [x] Add tests to cover changes as needed.
- [x] Update documentation as needed.

### CODE REVIEW

File: src/NEO/Address.cpp [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+    int size = base58_decode_check(string.data(), HASHER_SHA2D, buffer, (int)capacity);
+    if (size != Address::size || buffer[0] != version) {
+        return false;
+    }
+
+    return true;
+}
+
+Address::Address(const std::string& string) {
```
Comment (vikmeup) [inline src/NEO/Address.cpp]: This looks like a copy of Base58 address chain, did you consider reusing code from other chains?

Comment (ArtemGoryunov) [reply src/NEO/Address.cpp]: Yes, I'm going to use `std::string bytesToBase58(const uint8_t *data, size_t dataSize)` from Tezos

File: src/NEO/Address.cpp [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    size_t capacity = 128;
+    uint8_t buffer[capacity];
+
+    int size = base58_decode_check(string.data(), HASHER_SHA2D, buffer, (int)capacity);
+    assert(size == Address::size);
```
Comment (alejandro-isaza) [inline src/NEO/Address.cpp]: Let's not use assert, look at the latest code on master.

File: src/NEO/Address.cpp (hunk 1 above)
Comment (vikmeup) [reply src/NEO/Address.cpp]: @alejandro-isaza can you advise how should we move forward with this? there is only one byte that is different, any way to derivate base class?

Comment (alejandro-isaza) [reply src/NEO/Address.cpp]: Yes, we should definitely refactor. I can add this to my todo list.

File: src/NEO/Address.cpp (hunk 2 above)
Comment (ArtemGoryunov) [reply src/NEO/Address.cpp]: done

File: src/NEO/Address.cpp (hunk 1 above)
Comment (ArtemGoryunov) [reply src/NEO/Address.cpp]: done

**Comment by ArtemGoryunov** [comment]
> Looks good. Remember to add to Coin.cpp once we have a CoinType

yes, I'm working on it.

**Pull request merged**